src/browser/progressbar.js

```diff
@@ -18,6 +18,42 @@ define((require, exports, module) => {
   const {Element, VirtualAttribute} = require('./element');
   const Component = require('omniscient');
 
+  // Animation parameters:
+  const A = 0.2;              // Zone A size (a full progress is equal to '1'
+  const B = 0.2;              // Zone B size
+  const APivot = 200;         // When to reach ~80% of zone A
+  const BPivot = 500;         // When to reach ~80% of zone B
+  const CDuration = 200;     // Time it takes to fill zone C
+  const ApproachFunc = (tMs, pivoMs) => 2 * Math.atan(tMs / pivoMs) / Math.PI;
+
+  function ComputeProgress(viewer) {
+    // Inverse tangent function: [0 - inf] -> [0 - PI/2]
+    // ApproachFunc: [time, pivot] -> [0 - 1]
+    // Pivot value is more or less when the animation seriously starts to slow down
+    let progress = 0; // value between 0 and 1
+
+    let now = performance.now();
+
+    if (viewer.get('readyState')) { // Not an empty viewer
```

Comment:
Suggestion: Maybe we should have a function to check if viewer isn't empty as in some places we check for `uri` field here for `readyState` I think that way we may end up causing regressions any time any part of data is changed.